src/trio/_sync.py

```diff
@@ -571,11 +572,19 @@ def acquire_nowait(self) -> None:
         """
 
         task = trio.lowlevel.current_task()
+
         if self._owner is task:
             raise RuntimeError("attempt to re-acquire an already held Lock")
         elif self._owner is None and not self._lot:
             # No-one owns it
             self._owner = task
+        elif (
+            self._owner is not None
+            and inspect.getcoroutinestate(self._owner.coro) is inspect.CORO_CLOSED
+        ):
+            raise trio.BrokenResourceError(
+                "Attempted acquire on Lock which will never be released, owner has exited.",
```

Comment:
```suggestion
                "Attempted acquire on Lock which will never be released as the owner has exited.",
```

I'm not sure this is the ideal message either way but the alternative I can think of isn't great either: "Deadlock detected: acquiring lock which will never be released."